presto-jmx/src/main/java/com/facebook/presto/connector/jmx/JmxConnector.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.connector.jmx;
+
+import com.facebook.presto.spi.connector.Connector;
+import com.facebook.presto.spi.connector.ConnectorTransactionHandle;
+import com.facebook.presto.spi.transaction.IsolationLevel;
+import com.google.inject.Inject;
+import io.airlift.log.Logger;
+
+import static com.facebook.presto.spi.transaction.IsolationLevel.READ_COMMITTED;
+import static com.facebook.presto.spi.transaction.IsolationLevel.checkConnectorSupports;
+import static java.util.Objects.requireNonNull;
+
+public class JmxConnector
+        implements Connector
+{
+    public static final String CONNECTOR_ID_PARAMETER = "jmx.connectorId";
+
+    private static final Logger log = Logger.get(JmxConnector.class);
+
+    private final JmxMetadata jmxMetadata;
+    private final JmxHistoryDumper jmxHistoryDumper;
+    private final JmxSplitManager jmxSplitManager;
+    private final JmxRecordSetProvider jmxRecordSetProvider;
+
+    @Inject
+    public JmxConnector(
+            JmxMetadata jmxMetadata,
+            JmxSplitManager jmxSplitManager,
+            JmxRecordSetProvider jmxRecordSetProvider,
+            JmxHistoryDumper jmxHistoryDumper)
+    {
+        this.jmxMetadata = requireNonNull(jmxMetadata, "jmxMetadata is null");
+        this.jmxSplitManager = requireNonNull(jmxSplitManager, "jmxSplitManager is null");
+        this.jmxRecordSetProvider = requireNonNull(jmxRecordSetProvider, "jmxRecordSetProvider is null");
+        this.jmxHistoryDumper = requireNonNull(jmxHistoryDumper, "jmxHistoryDumper is null");
+        jmxHistoryDumper.start();
```

Comment:
Why does this class manage JmxHistoryDumper's lifecycle? Instead, tag that class's `start()` with @PostConstruct and the injection framework will do it.